nixos/modules/services/backup/syncoid.nix

```diff
@@ -0,0 +1,129 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.syncoid;
+in {
+
+    # Interface
+
+    options.services.syncoid = {
+      enable = mkEnableOption "Syncoid ZFS synchronization service";
+
+      interval = mkOption {
+        type = types.str;
+        default = "hourly";
+        example = "*-*-* *:15:00";
+        description = ''
+          Run syncoid at this interval. The default is to run hourly.
+
+          The format is described in
+          <citerefentry><refentrytitle>systemd.time</refentrytitle>
+          <manvolnum>7</manvolnum></citerefentry>.
+        '';
+      };
+
+      user = mkOption {
+        type = types.str;
+        default = "root";
+        example = "backup";
+        description = ''
+          The user for the service. ZFS privilege delegation must be configured
+          to use a user other than root.
+        '';
+      };
+
+      sshKey = mkOption {
+        type = types.nullOr types.path;
+        default = null;
+        description = ''
+          SSH private key file to use to login to remote system. Can be
+          overridden in individual commands.
+        '';
+      };
+
+      defaultArguments = mkOption {
+        type = types.separatedString " ";
+        default = "";
+        example = "--no-sync-snap";
+        description = ''
+          Arguments to add to every syncoid command, unless disabled for that
+          command.
+        '';
+      };
+
+      commands = mkOption {
+        type = types.listOf (types.submodule {
```

Comment:
I recommend `types.attrsOf` instead of `listOf`, since lists can't be changed after they're defined. By assigning an attribute name to entries this is possible.